### DISCUSSION TITLE
Execute matches from command line without a trigger word (or without backspacing the trigger word)

### DISCUSSION THREAD

**Discussion by bekoeppel** [espanso/espanso#1932] [Feature requests and Ideas]
I have a few matches which I would like to execute based on various keyboard shortcuts (incl. Ctrl, Alt, Meta keys). Currently Espanso doesn't support such keyboard shortcuts, so I would like to trigger them from i3, like so:

```
bindsym $mod+Shift+D exec espanso match exec -t :sometrigger
``` 

As far as I found, I need to refer to the match which I'd like to execute by its trigger word. However, this leads to the situation that Espanso performs the match, and then sends 12 backspace characters, to erase the `:sometrigger` trigger word. Since I've triggered the match from the command line, the 12 backspaces remove something else, as I didn't actually type the trigger word.

I see the following possible solutions:
  1. Of course adding support for _any_ key binding to trigger a match would be ideal
  1. Somehow allow identifying matches from the command line without specifying the trigger (maybe by an `id`?)
  2. Add a command line flag to indicate whether I want the backspaces, or not

**Comment by smeech** [comment]
Espanso allows the use of some \<ctrl+keys> because they are associated with [control "characters"](https://espanso.org/docs/matches/basics/#keyboard-triggers). Extension of this to \<meta> and \<alt> combinations is a much requested feature but would require a major rewrite to Espanso, I'm afraid.

**Reply by bekoeppel** [reply]
I assumed that it would be complex to add Meta and Alt, so I'm more interested in my bullets 2 + 3

**Comment by smeech** [comment]
Further to the above, the first workaround I've found to your problem, is to invoke the search bar, and trigger the expansion from there. I agree, it would be nice to have a flag to `espanso match exec -t` which cancels any backspace deletion of the (non-existent) trigger text.

A small helper script:
```bash
#!/bin/bash
espanso cmd search
xdotool sleep 0.2 type "$1"
xdotool key Return
```
invoked from a key combination as e.g., `test.sh :espanso` will trigger the expansion from your hotkey without deletions.

**Reply by smeech** [reply]
Forget that!

Just bind the key to `xdotool sleep 0.1 type :espanso` or whatever.

Obvious really!

**Reply by bekoeppel** [reply]
Ah, yes! Good thought :+1:

**Comment by smeech** [comment]
#1933
